web-nfc/NFCWriter_push.https.html

```diff
@@ -75,81 +75,98 @@
     ];
 
 promise_test(t => {
-  let promises = [];
+  const writer = new NFCWriter();
+  const promises = [];
   invalid_type_messages.forEach(message => {
     promises.push(
-      promise_rejects(t, new TypeError(), navigator.nfc.push(message)));
+      promise_rejects(t, new TypeError(), writer.push(message)));
   });
   return Promise.all(promises);
-}, "Test that promise is rejected with TypeError if NDEFMessage is invalid.");
+}, "Test that promise is rejected with TypeError if NDEFMessageSource is invalid.");
 
 promise_test(t => {
-  let promises = [];
+  const writer = new NFCWriter();
+  const promises = [];
   invalid_syntax_messages.forEach(message => {
     promises.push(
-      promise_rejects(t, 'SyntaxError', navigator.nfc.push(message)));
+      promise_rejects(t, 'SyntaxError', writer.push(message)));
   });
   return Promise.all(promises);
 }, "'Test that promise is rejected with SyntaxError if NDEFMessage contains invalid records.");
 
 promise_test(t => {
-  return promise_rejects(t, new TypeError(), navigator.nfc.push(test_text_data, { timeout: "invalid"}));
-}, "nfc.push should fail with TypeError when invalid timeout is provided.");
+  const writer = new NFCWriter();
+  const controller = new AbortController();
+  const p = writer.push(test_text_data, { signal: controller.signal })
+  controller.abort();
+  return promise_rejects(t, 'NoModificationAllowedError', p);
+}, "NFCWriter.push should fail if the instance has already initiated NFC data transfer.");
 
 promise_test(t => {
-  return promise_rejects(t, new TypeError(), navigator.nfc.push(test_text_data, { timeout: -1 }));
-}, "nfc.push should fail with TypeError when invalid negative timeout value is provided.");
+  const writer = new NFCWriter();
+  return promise_rejects(t, new TypeError(), writer.push(test_text_data, { timeout: "invalid"}));
+}, "NFCWriter.push should fail with TypeError when invalid timeout is provided.");
 
 promise_test(t => {
-  return promise_rejects(t, 'TimeoutError', navigator.nfc.push(test_text_data, { timeout: 1 }));
-}, "nfc.push should fail with TimeoutError when timer expires.");
+  const writer = new NFCWriter();
+  return promise_rejects(t, new TypeError(), writer.push(test_text_data, { timeout: -1 }));
+}, "NFCWriter.push should fail with TypeError when invalid negative timeout value is provided.");
 
 promise_test(t => {
-  let promise = navigator.nfc.push(test_text_data, { timeout: 100 });
-  navigator.nfc.cancelPush();
-  return promise_rejects(t, 'AbortError', promise);
-}, "nfc.cancelPush should reject pending promise with AbortError.");
+  const writer = new NFCWriter();
+  return promise_rejects(t, 'TimeoutError', writer.push(test_text_data, { timeout: 1 }));
+}, "NFCWriter.push should fail with TimeoutError when timer expires.");
 
 promise_test(t => {
```

Comment:
Nit: Could you move this test behind the other `Signal` test above? Just to make tests ordered. :)